valhalla/baldr/graphtile.h

```diff
@@ -181,6 +181,41 @@ class GraphTile {
         " nodecount= " + std::to_string(header_->nodecount()));
   }
 
+  /**
+   * Get an OSM id for a node.
+   * @param node The GraphId of the node.
+   * @return  Returns the OSM id for the node.
+   */
+  uint64_t osmid_for_node(const GraphId& node) const {
+    assert(node.Tile_Base() == header_->graphid().Tile_Base());
+    return osmid_for_node(node.id());
+  }
+
+  /**
+   * Get an OSM id.
+   * @param  idx  Index of the node within the current tile.
+   * @return  Returns the OSM id.
+   */
+  uint64_t osmid_for_node(const size_t idx) const {
+    assert(has_osmids_for_nodes());
+    if (idx < header_->nodecount()) {
+      return osmids_for_nodes_[idx];
+    }
+    throw std::runtime_error(
+        std::string(__FILE__) + ":" + std::to_string(__LINE__) +
+        " GraphTile NodeInfo index out of bounds: " + std::to_string(header_->graphid().tileid()) +
+        "," + std::to_string(header_->graphid().level()) + "," + std::to_string(idx) +
+        " nodecount= " + std::to_string(header_->nodecount()));
+  }
```

Comment:
i think this bit is too strong tbh. it assumes the caller knows how the data was made (with or without node ids) or that they conditionally call the method. i'd prefer to return a sentinal value that the caller can check to see if the node had or didnt have an id. i know its all or nothing since it doesnt support sparse data but the assertion is just too much

```suggestion
  uint64_t osmid_for_node(const size_t idx) const {
    if (idx > header_->nodecount()) {
      throw std::runtime_error(
        std::string(__FILE__) + ":" + std::to_string(__LINE__) +
        " GraphTile NodeInfo index out of bounds: " + std::to_string(header_->graphid().tileid()) +
        "," + std::to_string(header_->graphid().level()) + "," + std::to_string(idx) +
        " nodecount= " + std::to_string(header_->nodecount()));
    }
    return header_->has_osmids_for_nodes() ? osmids_for_nodes_[idx] : kInvalidNodeId;
  }
```